controllers/openstackcontrolplane_controller.go

```diff
@@ -989,3 +813,112 @@ func (r *OpenStackControlPlaneReconciler) createOrUpdateOpenStackClient(
 
 	return osc, nil
 }
+
+func (r *OpenStackControlPlaneReconciler) ensureVIPs(
+	instance *ospdirectorv1beta1.OpenStackControlPlane,
+	cond *ospdirectorv1beta1.Condition,
+) (ctrl.Result, error) {
+	//
+	// Create VIPs for networks where VIP parameter is true
+	//
+
+	// create list of networks where Spec.VIP == True
+	vipNetworksList, err := r.createVIPNetworkList(instance, cond)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	var ctrlResult ctrl.Result
+	currentLabels := instance.DeepCopy().Labels
+	//
+	// add osnetcfg CR label reference which is used in the in the osnetcfg
+	// controller to watch this resource and reconcile
+	//
+	instance.Labels, ctrlResult, err = openstacknetconfig.AddOSNetConfigRefLabel(r, instance, cond, vipNetworksList[0])
```

Comment:
Could there ever be a need to loop through all networks in `vipNetworksList` here?  Just curious as to why the first network in the list and nothing else.

Reply:
no, unless we ever support having multiple osnetcfgs. This network is only used to identify the osnetcfg object.